### DISCUSSION TITLE
Welcome to sn_esp32 Discussions!

### DISCUSSION THREAD

**Discussion by wayne-love** [wayne-love/sn_esp32#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by matt-nz** [comment]
Hello,
Recently received my spa with spa net controller and stumbled across this page and the gist with comments on it looks like at least three of you are working on an implementation to control spanet without needing the offical module. I am a home assistant user and a software engineer by trade but mostly Node, Java with bits of other things. I've done a bit of diy with ESP8266s around the home and have a couple on hand so going to start with https://github.com/shanekuz/SPANETMCU. 

But also looking at https://raw.githubusercontent.com/wayne-love/sn_esp32/master/circuit/circuit.png could you please explain what you use to connect to the spa controller RJ45 port I presume on the two pin 6's? Also you are powering directly from the spa controller on pin 7 and 8?

**Reply by wayne-love** [reply]
> But also looking at https://raw.githubusercontent.com/wayne-love/sn_esp32/master/circuit/circuit.png could you please explain what you use to connect to the spa controller RJ45 port I presume on the two pin 6's? Also you are powering directly from the spa controller on pin 7 and 8?

@matt-nz Thanks for picking up the error with the circuit diagram, this has been fixed.  I've also added some images of the board that I've built.  The connector to the spa is a RJ45 socket that is mounted to the board with one of these https://core-electronics.com.au/breakout-board-for-rj45.html.  Then it is a standard CAT5 lead to connect to the port on the SpaNet controller.

**Reply by wayne-love** [reply]
> Also you are powering directly from the spa controller on pin 7 and 8?

Yes, indirectly.  Pin 7 is approximately +12 (sometimes up to 14v), Pin 8 is ground.  I use a buck converter (AP63203) to drop this down to 3.3v.

**Reply by matt-nz** [reply]
Thanks @wayne-love, I got SPANETMCU going. Would be keen to try out your one but I'm not clear what is required to make it work on a ESP-8266, so may look at adding your home assistant discovery there instead.

For anyone as clueless as me a couple of gotchas I encountered.
* Sparkfun RJ45 breakout board has the pins backwards so pin 1 corresponded to pin 8 on the RJ45
* The breakout board also does not make a good connection without soldering the RJ45 connector on I was measuring the voltages and was often getting zero making me check my patch cable and wondering if I had to turn something on on the controller first etc.
* Don't attempt to ground to anything not directly connected to controller I tried to ground to metal and caused a 12 volt system overload
* My controller consistently gives out 15 volts on pin 8 will probably fry my NodeMCU so powering off a battery until my buck converter arrives

**Reply by wayne-love** [reply]
@matt-nz 

> Thanks @wayne-love, I got SPANETMCU going. Would be keen to try out your one but I'm not clear what is required to make it work on a ESP-8266, so may look at adding your home assistant discovery there instead.

In theory, it should work with some mods on an 8266 but you may have a problem with the single UART on it.  I say this only because I see other people mention that this is an issue.  In theory, we have only 1 serial connection when the system is live so it should work fine.  YMMV.
